build/upload_artifacts_gh.sh

```diff
@@ -1,12 +1,104 @@
 #!/bin/bash
```

Comment:
added the same upload script as infrastructure-agent to retry on network errors during package push to assets. We should create an GH action in the future for this script to be able to reuse it